plugins/micron/micron-utils.c

```diff
@@ -0,0 +1,276 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+/* Copyright (c) Micron, Inc 2024.
+ * 
+ * @file: micron-utils.h
+ * @brief: This module contains all the utilities needed for micron nvme plugin and other micron modules.
+ * @author: Chaithanya Shoba <ashoba@micron.com>
+ */
+
+#include "micron-utils.h"
+
+int hex_to_int(char c)
+{
+    if (c >= '0' && c <= '9')
+        return c - '0';
+    else if (c >= 'A' && c <= 'F')
+        return 10 + (c - 'A');
+    else if (c >= 'a' && c <= 'f')
+        return 10 + (c - 'a');
+    else
+        return -1; // Invalid character
+}
+
+char* hex_to_ascii(const char* hex) 
+{
+    int hexLength = strlen(hex);
+    char* text = NULL;
+
+    if (hexLength > 0) {
+        int symbolCount;
+        int oddHexCount = hexLength % 2 == 1;
+
+        if (oddHexCount)
+            symbolCount = (hexLength / 2) + 1;
+        else
+            symbolCount = hexLength / 2;
+
+        text = (char*)malloc(symbolCount + 1); // Allocate memory for the result
+
+        int lastIndex = hexLength - 1;
+        for (int i = lastIndex; i >= 0; --i) {
+            if ((lastIndex - i) % 2 != 0) {
+                int dec = 16 * hex_to_int(hex[i]) + hex_to_int(hex[i + 1]);
+                if (oddHexCount)
+                    text[i / 2 + 1] = dec;
+                else
+                    text[i / 2] = dec;
+            } else if (i == 0) {
+                int dec = hex_to_int(hex[0]);
+                text[0] = dec;
+            }
+        }
+
+        text[symbolCount] = '\0'; // Terminate the string
+    }
+
+    return text;
+}
+    
+unsigned char *read_binary_file(char *data_dir_path, const char *bin_path, long *buffer_size, int retry_count)
```

Comment:
The line is tool long and also the function itself is too long so needed to split.

Reply:
splitted.